### ISSUE TITLE
Export a sketch's canvas element/context

### ISSUE THREAD

**Issue by baptistebriel** [mattdesl/canvas-sketch#21]
Hey!

Digging into some sketches using THREE.js, I was thinking of using another canvas as a texture uniform sent in a RawShaderMaterial. Basically would like to draw some text on a canvas and be able to use it in the current sketch.

Exporting a sketch's canvas element would maybe make this possible?

```javascript
const canvasSketch = require('canvas-sketch')
const anotherSketch = require('./another-sketch')

const currentSketch = ({ context }) => {
  // set up THREE.js renderer, etc...
  // then grab the anotherSketch's canvas and create a texture
  const texture = new THREE.Texture(anotherSketch.canvas)
  // etc...

  return {
    resize ({ pixelRatio, viewportWidth, viewportHeight }) {
      anotherSketch.resize({ pixelRatio, viewportWidth, viewportHeight })
    },
    render ({ time, deltaTime }) {
      anotherSketch.render({ time, deltaTime })
    }
  }
}

canvasSketch(currentSketch, settings)
```

Would be awesome to take advantage of this module and not having to re-create another canvas and set-up resize events. Maybe having another option in the settings of the sketch to prevent the canvas to be added to the body and functions to be called automatically?
Cheers!

**Comment by mattdesl** [comment]
Ok, I just added a small patch to `canvas-sketch@0.0.21` that allows you to disable mounting a canvas. Example:

```js
const sketch = async () => {
  const otherManager = await canvasSketch(otherSketch, {
    // Disable keyboard shortcuts on sub-sketch
    hotkeys: false,
    // Do not mount sub-sketch to any parent DOM element
    parent: false
  });

  // Access props
  console.log(otherManager.props.canvas);

  ...
};

canvasSketch(sketch);
```

I've added two examples to show how this could work:

- [animated-three-text-canvas.js](https://github.com/mattdesl/canvas-sketch/blob/master/examples/animated-three-text-canvas.js) - Here we pass new props through the settings object, so `manager.update({ text })` will trigger a re-render
- [canvas-in-canvas.js](https://github.com/mattdesl/canvas-sketch/blob/master/examples/canvas-in-canvas.js) - This is a bit more tricky/hacky because it tries to fit the canvas to the text size. We have to measure the text, then set the size, and then re-render. If you use `manager.update()` for all of that you will run into stack overflow

In all honesty, I think it's a bit overkill and will probably cause more headache than its worth. You might be better off just creating & sizing a new canvas element directly within your ThreeJS sketch. The hardest parts that `canvas-sketch` does is things like render loop, animation timing, physical unit scaling, CSS style retina scaling etc and if you don't need those things then its pretty straight-forward:

```js
const otherCanvas = document.createElement('canvas');
const otherContext = otherCanvas.getContext('2d');

// Set a size
otherCanvas.width = myWidth;
otherCanvas.height = myHeight;

// Now draw into canvas with otherContext
...
```

But it's a cool proof of concept, and I think worth making the library support this sort of thing. :smile:
